### ISSUE TITLE
Cannot handle file names with special characters properly

### ISSUE THREAD

**Issue by Amazingkenneth** [agrawal-d/cph#425]
Under the same compilation environment, files with character `Σ` in the file name automatically created by Competitive Companion cannot be compiled and tested properly.

Pop-up error message: `Could not launch testcase process. Is 'g++' in your PATH?`

The prob file **.cph/.C_Σ.cpp_dff3a7f9af8ab0b10eadb767a68213c6.prob**
```
{"name":"C - Σ","group":"AtCoder - UNIQUE VISION Programming Contest 2024 Spring(AtCoder Beginner Contest 346)","url":"https://atcoder.jp/contests/abc346/tasks/abc346_c","interactive":false,"memoryLimit":1024,"timeLimit":2000,"tests":[{"input":"4 5\n1 6 3 1\n","output":"11\n","id":1711195657300},{"input":"1 3\n346\n","output":"6\n","id":1711195657232},{"id":1711195657231,"input":"10 158260522\n877914575 24979445 623690081 262703497 24979445 1822804784 1430302156 1161735902 923078537 1189330739","output":"12523196466007058"}],"testType":"single","input":{"type":"stdin"},"output":{"type":"stdout"},"languages":{"java":{"mainClass":"Main","taskClass":"C"}},"batch":{"id":"abf60f90-8b57-4e1d-9daa-9aef40879756","size":1},"srcPath":"d:\\contest\\C_Σ.cpp"}
```

<img width="770" alt="Successful running testcases" src="https://github.com/agrawal-d/cph/assets/81886982/35ffbb1d-8b50-4ce2-8b62-beb02253036d">

<img width="770" alt="Cannot handle the filename" src="https://github.com/agrawal-d/cph/assets/81886982/6918048b-888c-4e4f-96a4-571d4c7d0320">

---

**Extension Version:** `v6.1.0`

**VS Code Version:** `1.87.2`

**Browser Version:** `Microsoft Edge - Version 122.0.2365.92 (Official build) (64-bit)`

**Operating System:** `Microsoft Windows 11 Home 10.0.22631`

**Comment by ritesh-tiw** [comment]
![image](https://github.com/agrawal-d/cph/assets/107099010/3926daa9-dae2-43cf-b679-57f6a746c4a2)
Hii I am not able to reproduce this issue on my side .
VS Code Version : v1.87.2
Extension Version: v6.1.0
Operating System: Windows 11

**Comment by Amazingkenneth** [comment]
I met the error when compiling using the [mingw](https://scoop.sh/#/apps?q=mingw&id=965b35870a2cbc6584a302ef1721c7f18faaef24) compiler in the package manager Scoop. The `bin` directory `D:\Scoop\apps\mingw\current\bin` has been added to the system environment variables.

---
Executables and Library files found in the directory:
<details>

```powershell
PS D:\Scoop\apps\mingw\current\bin> ls


    Directory: D:\Scoop\apps\mingw\current\bin


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a----        2023-10-17      3:16        1178112 addr2line.exe
-a----        2023-10-17      3:16        1205248 ar.exe
-a----        2023-10-17      3:16        1817600 as.exe
-a----        2023-10-17      6:07        2618880 c++.exe
-a----        2023-10-17      3:16        1175552 c++filt.exe
-a----        2023-10-17      6:07        2616832 cpp.exe
-a----        2023-10-17      3:16        1234432 dlltool.exe
-a----        2023-10-17      3:16          75776 dllwrap.exe
-a----        2023-10-17      3:16        2499584 dwp.exe
-a----        2023-10-17      3:16          63488 elfedit.exe
-a----        2023-10-17      6:07        2618880 g++.exe
-a----        2023-10-17      6:07          61952 gcc-ar.exe
-a----        2023-10-17      6:07          61952 gcc-nm.exe
-a----        2023-10-17      6:07          61952 gcc-ranlib.exe
-a----        2023-10-17      6:07        2615296 gcc.exe
-a----        2023-10-17      6:07        1863168 gcov-dump.exe
-a----        2023-10-17      6:07        1861632 gcov-tool.exe
-a----        2023-10-17      6:07        2040320 gcov.exe
-a----        2023-10-17      7:23           4587 gdb-add-index
-a----        2023-10-17      7:23          59684 gdb.exe
-a----        2023-10-17      7:23        9945600 gdborig.exe
-a----        2023-10-17      7:23         400896 gdbserver.exe
-a----        2023-10-17      6:09          60416 gendef.exe
-a----        2023-10-17      6:09          74240 genidl.exe
-a----        2023-10-17      6:09          31232 genpeimg.exe
-a----        2023-10-17      6:07        2617856 gfortran.exe
-a----        2023-10-17      3:16        1242112 gprof.exe
-a----        2023-10-17      3:16        1746944 ld.bfd.exe
-a----        2023-10-17      3:16        1746944 ld.exe
-a----        2023-10-17      3:16        4756480 ld.gold.exe
-a----        2023-10-17      6:08          32768 libatomic-1.dll
-a----        2023-10-17      6:08         111616 libgcc_s_seh-1.dll
-a----        2023-10-17      6:08        3118080 libgfortran-5.dll
-a----        2023-10-17      6:08         280576 libgomp-1.dll
-a----        2023-10-17      6:08         362496 libquadmath-0.dll
-a----        2023-10-17      6:08        2246144 libstdc++-6.dll
-a----        2023-10-17      6:08          54784 libwinpthread-1.dll
-a----        2023-10-17      6:07       32673280 lto-dump.exe
-a----        2023-10-17      7:24         270336 make.exe
-a----        2023-10-17      7:24         270336 mingw32-make.exe
-a----        2023-10-17      3:16        1194496 nm.exe
-a----        2023-10-17      3:16        1312256 objcopy.exe
-a----        2023-10-17      3:16        2415104 objdump.exe
-a----        2023-10-17      3:16        1205248 ranlib.exe
-a----        2023-10-17      3:16        1002496 readelf.exe
-a----        2023-10-17      3:16        1178112 size.exe
-a----        2023-10-17      3:16        1182208 strings.exe
-a----        2023-10-17      3:16        1312256 strip.exe
-a----        2023-10-17      6:10         530432 widl.exe
-a----        2023-10-17      3:16        1200640 windmc.exe
-a----        2023-10-17      3:16        1290752 windres.exe
-a----        2023-10-17      6:07        2618880 x86_64-w64-mingw32-c++.exe
-a----        2023-10-17      6:07        2618880 x86_64-w64-mingw32-g++.exe
-a----        2023-10-17      6:07        2615296 x86_64-w64-mingw32-gcc-13.2.0.exe
-a----        2023-10-17      6:07          61952 x86_64-w64-mingw32-gcc-ar.exe
-a----        2023-10-17      6:07          61952 x86_64-w64-mingw32-gcc-nm.exe
-a----        2023-10-17      6:07          61952 x86_64-w64-mingw32-gcc-ranlib.exe
-a----        2023-10-17      6:07        2615296 x86_64-w64-mingw32-gcc.exe
-a----        2023-10-17      6:07        2617856 x86_64-w64-mingw32-gfortran.exe


PS D:\Scoop\apps\mingw\current\bin>
```

</details>

**Comment by ritesh-tiw** [comment]
I have seen this issue happening very rarely ,when clicking run option more than one time consecutively.
Is it still reproducible on your side ??

**Comment by Amazingkenneth** [comment]
Yes, but the error is only on the faulty file, and the other files compile normally.

**Comment by Amazingkenneth** [comment]
**Solution:**
The issue was resolved by setting the system locale to UTF-8, which allows the linker (`ld.exe`) to properly handle non-ASCII characters in file names.

**Steps to Set System Locale to UTF-8 on Windows:**
1. Open the Control Panel.
2. Go to "Region" and then the "Administrative" tab.
3. Click "Change system locale..."
4. Modify "Current system locale" to a character set that contains the character (for example, I chose Simplified Chinese) and uncheck "Beta: Use Unicode UTF-8 for worldwide language support".
5. Restart your computer.

![Step by step](https://github.com/agrawal-d/cph/assets/81886982/5b0aeebe-143e-4e2f-b355-c0a5b2dc74b3)

After setting the system locale to UTF-8, the compilation and testing of files with non-ASCII characters in their names work correctly.
